src/gui/src/histogramWidget.cpp

```diff
@@ -0,0 +1,242 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (c) 2019, The Regents of the University of California
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice, this
+//   list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#include "histogramWidget.h"
+
+#include <QWidget>
+#include <QFrame>
+#include <QColor>
+#include <QString>
+#include <QtCharts>
+#include <vector>
+#include <cmath>
+#include <algorithm>
+#include <iostream>
+#include <set>
+
+#include <sta/MinMax.hh>
+#include "staGuiInterface.h"
+
+namespace gui {
+
+const int SELECT = 0;
+const int SLACK_MODE = 1;
+
+HistogramWidget::HistogramWidget(QWidget* parent)
+    : QDockWidget("Histogram Plotter", parent),
+      sta_(nullptr),
+      label_(new QLabel(this)),
+      mode_menu_(new QComboBox(this)),
+      chart_(new QChart),
```

Comment:
Does chart_ have a parent?

Reply:
The ownership of chart_ is passed to QChartView when the latter is initialized.
The order of ownership is `chart_`, `display_`, `layout`, `container` and finally the parent `QDockwidget`.

I just realized it would be more appropriate to initialize both QChart and QChartView like this:

```
display_(new QChartView(this)),
chart_(new QChart(display_))
```

Reply:
In QChart documentation, the constructor must be specified as:
`QChart::QChart(QGraphicsItem *parent = nullptr, Qt::WindowFlags wFlags = Qt::WindowFlags())`

There is no QGraphicsItem in the widget.

This is not a problem, because, when `display_` is initialized right after `chart_`, it takes its ownership.

To be more concise, I'll add a parent to `display_`, even though there is no leak, because `layout` takes its ownership in the constructor.